runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala

```diff
@@ -3,15 +3,48 @@
  */
 package akka.grpc
 
-import scala.concurrent.duration.Duration
+import akka.actor.ActorSystem
+import com.typesafe.config.Config
 
+import scala.concurrent.duration.Duration
 import io.grpc.CallCredentials
 
 object GrpcClientSettings {
   /** Scala API */
   def apply(host: String, port: Int): GrpcClientSettings =
     new GrpcClientSettings(host, port)
 
+  /** Scala API */
+  def apply(serviceName: String, sys: ActorSystem): GrpcClientSettings = {
+    val akkaGrpcClientConfig = sys.settings.config.getConfig("akka.grpc.client")
+
+    val serviceConfig =
+      if (akkaGrpcClientConfig.hasPath('"' + serviceName + '"'))
+        akkaGrpcClientConfig.getConfig('"' + serviceName + '"').withFallback(akkaGrpcClientConfig.getConfig("\"*\""))
+      else
+        akkaGrpcClientConfig.getConfig("\"*\"")
+
+    fromSubConfig(sys.settings.config, serviceConfig)
+  }
+
+  /** Java API */
+  def create(serviceName: String, sys: ActorSystem): GrpcClientSettings =
+    GrpcClientSettings(serviceName, sys)
+
+  def fromSubConfig(root: Config, c: Config): GrpcClientSettings = {
+    var result = GrpcClientSettings(c getString "host", c getInt "port")
+    if (c.hasPath("override-authority"))
```

Comment:
I'd prefer that we define those in reference conf (for documentation purposes) with default values such as `off`

Reply:
Done for the deadline. Do we also want this for string parameters? would the default be an empty string or some 'magic value'?